core/graph/src/main/java/uk/gov/gchq/gaffer/graph/Graph.java

```diff
@@ -912,22 +947,43 @@ private void updateSchema(final GraphConfig config) {
                         .build();
                 addSchema(newSchema);
             }
+
+            if (null != config.getLibrary() && config.getLibrary().exists(config.getGraphId())) {
+                // Set Schema if null.
+                final Pair<Schema, StoreProperties> pair = config.getLibrary().get(config.getGraphId());
+                schema = (null == schema) ? pair.getFirst() : schema;
+            }
         }
 
-        private void updateStoreProperties(final GraphConfig config) {
-            StoreProperties mergedStoreProperties = null;
+        private void updateStoreProperties(final GraphConfig config, final Store store) {
+            final StoreProperties mergedStoreProperties = new StoreProperties();
+
+            //Add Properties from ParentProperties
             if (null != parentStorePropertiesId) {
-                mergedStoreProperties = config.getLibrary().getProperties(parentStorePropertiesId);
+                mergedStoreProperties.merge(config.getLibrary().getProperties(parentStorePropertiesId));
             }
 
+            //Add Properties from Builder
             if (null != properties) {
-                if (null == mergedStoreProperties) {
-                    mergedStoreProperties = properties;
-                } else {
-                    mergedStoreProperties.merge(properties);
+                mergedStoreProperties.merge(properties);
+            }
+
+            //Add Properties from Store
+            if (null != store) {
```

Comment:
As a thought if you are expecting that a null `store` parameter may be passed to this method would it make more sense to overload this method instead so there is one with and one without out the `store` parameter?

Reply:
I think the timing and order is important of when the properties gets used. 
If I'm understanding you when overriding you can only add it before or after.  You also now have two methods and you still require the if statement to where it's called. More lines of code, not sure what polymorphism brings to the party in this situation. Its only used once and here in the builder.

Reply:
You can reach that line with null store like this

```java
 new Graph.Builder()
                .config(new GraphConfig.Builder()
                        .graphId("testGraph")
                        .build())
                .storeProperties(new MapStoreProperties())
                .addSchema(new Schema())
                .build();
```

Reply:
This has now been removed because it caused complexity with Kafka tests later down the line. these changes was reverted. but should be examined in the future.